fs/procfs/fs_procfsmeminfo.c

```diff
@@ -578,6 +578,13 @@ static int meminfo_stat(FAR const char *relpath, FAR struct stat *buf)
 
 void procfs_register_meminfo(FAR struct procfs_meminfo_entry_s *entry)
 {
+#ifdef CONFIG_BUILD_KERNEL
```

Comment:
@xiaoxiang781216 
Hmm, the change does not show Kmem.

the original PR

```
nsh> free
                   total       used       free    largest  nused  nfree
        Kmem:  125677760      20864  125656896  125651936     60      3
        Page:  134217728    2256896  131960832  131960832
```

after the changed applied

```
nsh> free
                   total       used       free    largest  nused  nfree
        Page:  134217728    2256896  131960832  131960832
```

Reply:
sorry the condition should be
if defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__)

Reply:
@xiaoxiang781216 
Hmm, it still does not work as expected.
The following is the call stack when Umem is registered.
As you can see `procfs_regster_meminfo` is linked to the kernel.

```
(gdb) where
#0  procfs_register_meminfo (entry=entry@entry=0x80200144) at procfs/fs_procfsmeminfo.c:582
#1  0x10805748 in mm_initialize (name=name@entry=0x1081beef "Umem", heapstart=0x80200150, heapstart@entry=0x80200000, heapsize=1048240) at mm_heap/mm_initialize.c:235
#2  0x10805da4 in umm_initialize (heap_start=heap_start@entry=0x80200000, heap_size=<optimized out>) at umm_heap/umm_initialize.c:84
#3  0x10814f30 in exec_module (binp=binp@entry=0x10827310, filename=filename@entry=0x1081b569 "/system/bin/init", argv=argv@entry=0x0, envp=0x10827420, envp@entry=0x0) at binfmt_execmodule.c:186
#4  0x10814420 in exec_spawn (filename=filename@entry=0x1081b569 "/system/bin/init", argv=argv@entry=0x0, envp=envp@entry=0x0, exports=exports@entry=0x0, nexports=nexports@entry=0, attr=attr@entry=0x10824dc8) \
at binfmt_exec.c:126
#5  0x108014e4 in nx_start_application () at init/nx_bringup.c:295
#6  nx_create_initthread () at init/nx_bringup.c:363
#7  nx_bringup () at init/nx_bringup.c:446
#8  0x1080143c in nx_start () at init/nx_start.c:691
#9  0x00000000 in ?? ()
(gdb) p entry->name
$3 = 0x1081beef "Umem"
```

Reply:
Got, the initialization is still happened in kernel space, not user space for umm.

Reply:
Yes.

Reply:
Umm is a special case, should we add both? mm_heap can be used by apps custom heap.

Reply:
@xiaoxiang781216 
I think I should modify `mm_initialize()` not to call `procfs_register_meminfo()` for the Umm instead of modifying `procfs_register_meminfo()`. What do you think?

Reply:
yes, it also possible to add bool argument to skip the register, or we can use name == NULL .

Reply:
OK, I'll try to modify them later.